src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -0,0 +1,143 @@
+
+object HullmodExoticHandler {
+    private val lookupMap: HashMap<HullmodExoticKey, HullmodExoticInstallData> = hashMapOf()
+
+    /**
+     * Method for checking whether the [HullmodExotic] should be installed onto a [ShipVariantAPI] by checking
+     * if we hadn't already installed it or not
+     *
+     * @param hullmodExotic the HullmodExotic in question
+     * @param parentFleetMember the [FleetMemberAPI] of the root module, or the whole ship's if it's a single-module ship
+     * @param variant the variant to check
+     * @param variantList list of module variants belonging to the parent module (optional)
+     */
+    fun shouldInstallHullmodExoticToVariant(hullmodExotic: HullmodExotic, parentFleetMember: FleetMemberAPI, variant: ShipVariantAPI, variantList: Optional<List<ShipVariantAPI>>): Boolean {
+        AnonymousLogger.log("--> shouldInstallHullmodExoticToVariant()\thullmodExotic: ${hullmodExotic}, variant: ${variant}", "HullmodExoticHandler")
+        // First things first, check if we have this in the map
+        // we will do this by forming up a [HullmodExotica, FleetMemberAPI] pair as a key
+        val hullmodExoticKey = HullmodExoticKey(
+                hullmodExotic = hullmodExotic,
+                parentFleetMember = parentFleetMember
+        )
+        val currentInstallData = lookupMap[hullmodExoticKey]
+
+        // now, if we already have some install data, we are going to be checking the variant we're about to install
+        // versus two different things:
+        // 1. is it in the expected list?
+        // 2. is it already NOT in the installed list?
+        // we'll just do nothing if it fails either of these and return false.
+        val retVal = if (currentInstallData != null) {
+            AnonymousLogger.log("shouldInstallHullmodExoticToVariant()\talready had install data !!!", "HullmodExoticHandler")
+            // now, check if this variant is in the expected list
+            val expectedVariants = currentInstallData.listOfExpectedVariants
+//            val isExpected: Boolean = expectedVariants.contains(variant)
+            // obviously these don't match
+            val isExpected: Boolean = expectedVariants
+                    .map { variant -> variant.hullSpec.hullId }
+                    .contains(variant.hullSpec.hullId)
+            AnonymousLogger.log("shouldInstallHullmodExoticToVariant()\tisExpected: ${isExpected}", "HullmodExoticHandler")
+
+            val variantsWeAlreadyInstalledOn = currentInstallData.listOfVariantsWeInstalledOn
+            val hasNotInstalledOnThisVariantAlready = variantsWeAlreadyInstalledOn.contains(variant).not()
+            AnonymousLogger.log("shouldInstallHullmodExoticToVariant()\thasNotInstalledOnThisAlready: ${hasNotInstalledOnThisVariantAlready}", "HullmodExoticHandler")
+
+            // Since I don't know how to check anything with parentFleetMember yet, lets just leave it out of the equation for now
```

Comment:
obviously this will become more relevant once i start testing with **two identical ships**.
basically do something like this:

```
val isForThisShip = currentInstallData.parentFleetMember == parentFleetMember
```
and just toss that in as the first eliminator condition.

BUT before that - add logging if we're getting hits on mismatched FleetMembers (aka ship2 trying to work with ship1's data)

Reply:
ah, i get it.
looking at the lookupMap definition, it's `HashMap<HullmodExoticKey, HullmodExoticInstallData>`
obviously, this means that *for each HullmodExoticKey there can be exactly one HullmodExoticInstallData* since keys are unique. or so I think.

Maybe this will need to become a `List<HullmodExoticInstallData>` but lets not rush too far ahead. Lets see how/when to toss in the parentFleetMember into the equation once I start testing with two ships.

Reply:
still waiting to see what happens with this

Reply:
comments here are still very relevant, even if the code they commented on no longer exists. this branch should really do the following and stress-test these:
- two identical ships installing/uninstalling
- one ship installing/uninstalling two different `HullmodExotic`s
- two identical ships installing/uninstalling two different `HullmodExotic`s

Reply:
so, resolving this for now.